File: Exercism/Go/space-age/space_age.go
package space

type Planet string

func Age(t float64, p Planet) float64 {
	multiplier := 1.0
	switch p {
	case "Mercury":
		multiplier = 0.2408467
	case "Venus":
		multiplier = 0.61519726
	case "Earth":
		multiplier = 1 // is this needed? since default is already set to 1
	case "Mars":
		multiplier = 1.8808158
	case "Jupiter":
		multiplier = 11.862615
	case "Saturn":
		multiplier = 29.447498
	case "Uranus":
		multiplier = 84.016846
	case "Neptune":
		multiplier = 164.79132
	default:
		multiplier = 1
	}
	return (t / 31557600 / multiplier)
}

/*
Stuff I needed
Arguments in functions: https://www.geeksforgeeks.org/function-arguments-in-golang/
Implementing a custom type: https://stackoverflow.com/questions/18992930/create-a-custom-type-with-constraint-in-go
*/

File: Exercism/cpp/space-age/space_age.cpp
#include "space_age.h"

using namespace std;
namespace space_age
{
	space_age::space_age(float seconds): time(seconds) {}

	float space_age::on_planet(string planet) const
	{
		return space_age_cache.at(planet);
	}
	float space_age::calculate_age_on_planet(string planet, float time) const{
		return time / 60 / 60 / 24 / 365.25 / multipliers.at(planet);
	}
	float space_age::on_mercury() const
	{
		return on_planet("Mercury");
	}
	float space_age::on_venus() const
	{
		return on_planet("Venus");
	}
	float space_age::on_earth() const
	{
		return on_planet("Earth");
	}
	float space_age::on_mars() const
	{
		return on_planet("Mars");
	}
	float space_age::on_jupiter() const
	{
		return on_planet("Jupiter");
	}
	float space_age::on_saturn() const
	{
		return on_planet("Saturn");
	}
	float space_age::on_uranus() const
	{
		return on_planet("Uranus");
	}
	float space_age::on_neptune() const
	{
		return on_planet("Neptune");
	}
	long space_age::seconds() const
	{
		return (long)time;
	}

	} // namespace space_age
